proto/gz/msgs/material_color.proto

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2024 CogniPilot Foundation
+ * Copyright (C) 2024 Rudis Laboratories LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package gz.msgs;
+option java_package = "com.gz.msgs";
+option java_outer_classname = "MaterialProtos";
+
+/// \ingroup gz.msgs
+/// \interface MaterialColor
+/// \brief Color description for an entities link-visual material.
+
+import "gz/msgs/color.proto";
+import "gz/msgs/header.proto";
+
+message MaterialColor
+{
+  /// \brief Optional header data
+  Header header          = 1;
+
+  /// \brief Name of link-visual.
+  string name            = 2;
```

Comment:
Hmmm, I guess that is already supported by the ros_gz bridge, and would be trivial I guess to add to the message.

My main goal behind this message is for simple "outside" control/use. I figure if you want to use this in a direct plugin to begin with might as well just use the VisualCmd service call directly. 

I'm not sure if there is any reasonable way to get the entity id hence why I excluded it.

Reply:
There are existing utility functions (eg. https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/Util.cc#L599) that understand the `Entity` message and I think the `name` field has a precedent of allowing `::` to be used to specify nested entities (https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/include/gz/sim/Util.hh#L70-L93). This would allow you to avoid name collisions better than a `name` and `parent_name` combination.

Reply:
If Entity msg is preferred for those I can change that.

Reply:
Thanks!

Reply:
Is there a reason why [`Visual` message does not use `Entity`](https://github.com/gazebosim/gz-msgs/blob/ce3ccb9af31ad93c580d5024cf130d373d632eb1/proto/gz/msgs/visual.proto#L47-L73)? I was following the way `Visual` message is already used in the code, where component inspector is already aware of the entities ID (as it is manually selected) and sends that as a service request: 
https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/gui/plugins/component_inspector_editor/ComponentInspectorEditor.cc#L1261-L1286

That same `Visual` message request gets listened on `visual_config` service topic which proceeds to the [callback on VisualService](https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/systems/user_commands/UserCommands.cc#L939-L954). That proceeds to `VisualCommand`:
https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/systems/user_commands/UserCommands.cc#L1716-L1783

For my work I wanted to replicate this behavior but externally like what was done for the `light_config` message topic (instead of only service) with callback on [`OnCmdLight` and then sends the subscribed message](https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/systems/user_commands/UserCommands.cc#L830-L841) to [`LightCommand`](https://github.com/gazebosim/gz-sim/blob/c6aaaf2d91a44641ab152143aa1f6cc9d7c0596f/src/systems/user_commands/UserCommands.cc#L172-L239)

To match that I [mapped the "condensed version" of `Visual` message in this case `MaterialColor` to the `Visual` message to be directly used in the above `VisualCommand`](https://github.com/gazebosim/gz-sim/blob/cfb0d74b0aadd7a2c9ddadbd5d7db234bd1d9316/src/systems/user_commands/UserCommands.cc#L969-L999) from the same style of [topic subscriber callback,`onCmdMaterialColor`](https://github.com/gazebosim/gz-sim/blob/cfb0d74b0aadd7a2c9ddadbd5d7db234bd1d9316/src/systems/user_commands/UserCommands.cc#L677-L679) as what was done with `OnCmdLight`.

Reply:
So I guess part of me would want to update `Visual` message to instead have `Entity` if that's the preferred route. As this is meant to be a specific `MaterialColor` changing only subset of the actual `Visual` message.

Reply:
All good, has entity now. 👼🏼